msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/B2CAuthority.java

```diff
@@ -26,27 +26,42 @@ class B2CAuthority extends Authority {
     }
 
     private void validatePathSegments(String[] segments) {
-        if (segments.length < 3) {
+        if (segments.length < 2) {
             throw new IllegalArgumentException(
-                    "B2C 'authority' Uri should have at least 3 segments in the path " +
-                            "(i.e. https://<host>/tfp/<tenant>/<policy>/...)");
+                    "Valid B2C 'authority' URLs should follow either of these formats: https://<host>/<tenant>/<policy>/... or https://<host>/something/<tenant>/<policy>/...");
         }
     }
 
     private void setAuthorityProperties() {
         String[] segments = canonicalAuthorityUrl.getPath().substring(1).split("/");
 
+        // In the early days of MSAL, the only way for the library to identify a B2C authority was whether or not the authority
+        //   had three segments in the path, and the first segment was 'tfp'. Valid B2C authorities looked like: https://<host>/tfp/<tenant>/<policy>/...
+        //
+        // More recent changes to B2C should ensure that any new B2C authorities have 'b2clogin.com' in the host of the URL,
+        //   so app developers shouldn't need to add 'tfp' and the first path segment should just be the tenant: https://<something>.b2clogin.com/<tenant>/<policy>/...
+        //
+        // However, legacy URLs using the old format must still be supported by these sorts of checks here and elsewhere, so for the near
+        //   future at least we must consider both formats as valid until we're either sure all customers are swapped,
+        //   or until we're comfortable with a potentially breaking change
         validatePathSegments(segments);
 
-        policy = segments[2];
-
-        final String b2cAuthorityFormat = "https://%s/%s/%s/%s/";
-        this.authority = String.format(
```

Comment:
It is better to keep the final constant out of the String.format method call. All the constants should ideally be declared separately.

Reply:
Originally I just got rid of the final entirely, because it didn't make sense to have a final in a method and have it only be used once in that method.

In the latest commit there are now constants for both the legacy and current formats (created outside the method). It's definitely more readable/maintainable to have constants for the two formats, even if they're only used once each.